Applications/Tizen_native/CustomShortcut/src/data.c

```diff
@@ -297,18 +322,108 @@ void data_train_model() {
     goto CLEAN_UP;
   }
 
-  freopen("out.txt", "a+", stdout);
-
   status = ml_train_model_run(model, NULL);
   if (status != ML_ERROR_NONE) {
     LOG_E("run model failed %d", status);
     goto CLEAN_UP;
   }
 
 CLEAN_UP:
+  // restore stdout
+  fflush(stdout);
+  dup2(fd_stdout, 1);
+  close(fd_pipe);
+
   status = ml_train_model_destroy(model);
   if (status != ML_ERROR_NONE) {
-    LOG_E("Destryoing model failed %d", status);
+    LOG_E("Destroying model failed %d", status);
+  }
+  return NULL;
+}
+
+int data_parse_result_string(const char *src, train_result_s *train_result) {
+  // clang-format off
+  // #10/10 - Training Loss: 0.398767 >> [ Accuracy: 75% - Validation Loss : 0.467543 ]
+  // clang-format on
+  int status = APP_ERROR_NONE;
+  int max_len = 512;
+  char buf[512];
+
+  regex_t pattern;
+  LOG_D(">>> %s", src);
+  char *pattern_string =
+    "#([0-9]+).*Training Loss: ([0-9]+[.][0-9]*)"
```

Comment:
Shouldnt training loss also be `([0-9]+\\.?[0-9]*)`?

Reply:
Fixed.